### PR TITLE
GH-37244: [Python] Remove support for pickle5

### LINKED ISSUE #37244: [Python] Remove support for pickle5
### Describe the enhancement requested

`pickle5` is a backport of pickle protocol 5 features from python 3.8.3 to python 3.5, 3.6, and 3.7.[1]

Now that python 3.7 is end-of-life and not supported in Arrow, we can remove all usage of it.

[1] https://pypi.org/project/pickle5/

### Component(s)

Python

### PR DESCRIPTION
**Pull request by chrisjordansquire**
Resolve issue https://github.com/apache/arrow/issues/37244 by removing pickle5 usage in pyarrow.

### Rationale for this change

See issue https://github.com/apache/arrow/issues/37244 . 

### What changes are included in this PR?

pickle5 usage is removed from pyarrow . 

### Are these changes tested?

Yes, the python test suite was run. 

### Are there any user-facing changes?

No. 

* Closes: #37244

### CODE REVIEW

**Comment by chrisjordansquire** [comment]
I believe the AppVeyor CI is failing due to an issue addressed in PR https://github.com/apache/arrow/pull/37558 . These tests were failing locally on my master as well, so I wouldn't expect the AppVeyor CI check to succeed until the above PR is merged in and I rebase.

**Comment by danepitkin** [comment]
LGTM! Thanks @chrisjordansquire! Adding a committer for final review.

**Review by AlenkaF — APPROVED** [review]
Thank you for the contribution @chrisjordansquire, much appreciated!

**Pull request merged**
